ingestion/src/metadata/mixins/pandas/pandas_mixin.py

```diff
@@ -80,50 +81,57 @@ def get_partitioned_df(self, dfs):
             for df in dfs
         ]
 
-    def return_ometa_dataframes_sampled(
+    def return_ometa_dataframes_sampled(  # pylint: disable=inconsistent-return-statements
         self, service_connection_config, client, table, profile_sample_config
     ):
         """
         returns sampled ometa dataframes
         """
         from pandas import DataFrame  # pylint: disable=import-outside-toplevel
 
-        data = ometa_to_dataframe(
-            config_source=service_connection_config.configSource,
-            client=client,
-            table=table,
-        )
-        if isinstance(data, DataFrame):
-            data: List[DataFrame] = [data]
-
-        if data and isinstance(data, list):
-            random.shuffle(data)
+        try:
+            data = ometa_to_dataframe(
+                config_source=service_connection_config.configSource,
+                client=client,
+                table=table,
+            )
+            if isinstance(data, DataFrame):
+                data: List[DataFrame] = [data]
+            if not isinstance(data, list):
+                raise TypeError(f"Couldn't fetch {table.name.__root__}")
```

Comment:
why is this in the if statement and not?

```python
if isinstance(<>):
    ...
if data and isinstance(<>):
    ...
raise TypeError(f"Couldn't fetch {table.name.__root__}")
```

You also have add `# pylint: disable=inconsistent-return-statements` that must tip you of that something is not concistent with the logic right?

Reply:
1st if: it's not a list of dataframes but just a dataframe, we convert it into list of dataframes just in case.
2nd if: if the data is not an empty list and check if it's of type list

inconsistent-return-statements is because we have not added a return to the exception

      except TypeError as err:
                  logger.debug(traceback.format_exc())
                  logger.error(err)

Reply:
That make sense. A few things:
1. My orginal comment simply asked to replace `return []` by `raise Something`. There is no point returning an empty list. This just makes things harder to debug. At this point if, for some reasons, we don't have a list of DataFrame then let's raise the error
2. I am not what the try/except is trying to catch here. From what I understand. If there is a failure here, it will be caught at the workflow level and raise there -- wouldn't it? Our try except here, would keep the program execution going until it fails on the next step because the variable calling this function would be `None` when we would expect it to be `list[df]` making the program slightly more difficult to debug.

Does it make sense?

Reply:
that makes it more clear.